github-management/kubernetes-repositories.md

```diff
@@ -127,6 +130,62 @@ ecosystem. Carries the endorsement of the Kubernetes community.
      Kubernetes steering committee elections. in the Kubernetes community
    * Repository must be approved by SIG-Architecture
 
+## Creating Repositories
+
+### Non-staging repositories
+
+For non-staging repositories, suggestions on how to create
+a new repository are described below.
+
+  * Ensure that the repo creation request has appropriate approvals
+  as per the rules mentioned above.
+  * Using the organization and repository name mentioned in the repo creation
+  request, create a new repo with default GitHub settings.
+  * Set the description as per the repo creation request.
+  * Clone the newly created repo locally.
+  * Copy the latest contents of [kubernetes-template-project] and create an
+  initial commit with the message *Initial commit from the
+  kubernetes-template-project*.
+  * Make the following changes in the next commit:
+    * If the request references a GitHub team to be listed in the `OWNERS`
+    file, update the `OWNERS_ALIASES` file to remove the steering-committee
+    alias and add a new alias for the team with members populated as per the
+    GitHub team. If the request does not reference a GitHub team, remove the
+    `OWNERS_ALIASES` file.
+    * Update the OWNERS file as per the request. If the repo is a
+    [SIG Repository], add a labels entry for the SIG that the repo belongs to.
+    * Update the `SECURITY_CONTACTS` file as per the request. Note that aliases
+    cannot be used in this case so expand the GitHub team, if specified.
+    * Create a new commit with the message *Update OWNERS, OWNERS_ALIASES and
+    SECURITY_CONTACTS*.
+  * Push the new commits directly to the master branch.
+  * If the repo is a [SIG Repository], add a new topic of the form
+  `k8s-sig-<sig-name-repo-belongs-to>` using the *Manage Topics* option.
+  * Create a PR against [kubernetes/org] to add teams as per the [team guidance]
+  for alloting repo admin and write access.
+  * Once the above PR is merged and the postsubmit has run, the new GitHub teams
+  will be created. In the *Collaborators and Teams* section in Settings,
+  assign the new teams appropriate access to the repo.
+  * Ask the author of the repo creation request to add the repo
+  as a part of a subproject in [`sigs.yaml`](/sigs.yaml).
+
+### Staging Repositories
+
+If the repository is a staging repository, there are some deviations
+from the above procedure:
+
+  * The repository **must** have an initial empty commit. The contents of the
+  repo will be populated from staging by the [publishing-bot].
+  * Setup branch protection and enable access to the
+  `stage-bots` team by adding the repo in
+  [`prow/config.yaml`](https://git.k8s.io/test-infra/prow/config.yaml). See
+  [kubernetes/test-infra#9292](https://github.com/kubernetes/test-infra/pull/9292)
+  for an example.
+  * Once the repo has been created, ask [@nikhita] to update the
+  [publishing-bot] configuration to include the repo.
```

Comment:
I'm not sure if this is ok? Should I be mentioning someone else here as well?

Reply:
Backing up your comment, I would put a team reference, a step like 'create an issue in publishing-bot repo' or link to the [publishing-bot playbook](https://github.com/kubernetes/publishing-bot#playbook) directly. Don't want to introduce a SPOF if at all possible.

Reply:
> Don't want to introduce a SPOF if at all possible.

Yeah, that was my idea as well. :+1: 

> I would put a team reference, a step like 'create an issue in publishing-bot repo' or link to the [publishing-bot playbook](https://github.com/kubernetes/publishing-bot#playbook) directly.

I'd be fine with either. @sttts any preferences?

Reply:
I like both. There's currently nothing in the playbook around how a non-publishing bot admin would add/request a repo be added.

Suggesting:
- Notes in the playbook on requesting a new repo to be added to publishing
- An issue template in k/publishing-bot to guide someone on how to request an addition
- Reference a GH team or [k/publishing-bot OWNERS](https://github.com/kubernetes/publishing-bot/raw/master/OWNERS) in this documentation, instead of an individual